File: Interview and LeetCode problems/twoSum.java
import java.util.*;
	
	// array [1, 2, 3, 7, 6]
	//t = 9
	//given an array and target number,  return indices of one pair that adds up to the 
//number in the array

public class twoSum {
	public static int[] returnIndices (int t, int[] args){
		HashMap<Integer, Integer> hashTable = new HashMap<Integer, Integer>();
		int array[] = new int [2];
		for (int i = 0; i < args.length; i++){
			hashTable.put(args[i], i);
		}//putting in for loop

		for (int i = 0; i < args.length; i++){
			int difference = t - args[i];
			if (hashTable.containsKey(args[i])){
				array[0] = hashTable.get(difference);
				array[1] = i;
				return array;
			}//if
		}//for

		array[0] = -1;
		return array;
	}//blah
	public static void main (String[] args){
		int returnArr[] = new int[2];
		int testArr[] = new int[]{1,2,6,3,7,8};

		returnArr = returnIndices(9, testArr);
		System.out.println(returnArr[0]);
		System.out.println(returnArr[1]);

		//Hash array into a hashmap
		//key 
	}
}//twoSum

File: MiscInterview/twoSum.py
#array [1, 2, 3, 7, 6]
#t = 9
#given an array and target number,  return indices of one pair that adds up to the 
#number in the array

def twoSum (t, arr =[]):
	hashTable = dict()

	for i in range(len(arr)):
		hashTable[arr[i]] = i

	for i in range(len(arr)):
		difference = t-arr[i]

		if difference in hashTable:
			print hashTable[difference] 
			print hashTable[arr[i]]
			break

	print hashTable

#O(n)
l = [1,2,3,7,6]
twoSum (9, l)

#implementation below prints duplicates and if target/2 exists, then removes that possibility too
#/* Given a list of numbers and numbers and print all pairs of the numbers from the list which adds up to given number 
#Input : {1,3,4,5,6,2,2} 6
#output: 
#1,5
#2,4
#2,4
#*/#
#
#

#def numSum (arr, t):
#    hashMap = dict()
#    count = 1 
#       
#    for i in range(0, len(arr)-1):
#       if arr[i] in hashMap:
#           count += 1
#           hashMap[arr[i]] = count
#           
#       elif arr[i] not in hashMap:
#           hashMap[arr[i]] = 1 
#      
#    for j in range(0, len(arr)-1):      
#        difference = t - arr[j]
#        
#        if arr[j] == difference and hashMap[difference] == 1:
#            continue
#        
#        #two conditions
#        #if t/2 occurs in input, and only occurs once, don't print
#        #if difference occurs more than once as count, then print those duplicates#

#        if difference in hashMap:
#            if hashMap[difference] > 1:
#                for k in range(0, hashMap[difference]):
#                    print difference
#                    print arr[j]
#                    continue
#            print difference
#            print arr[j]
#            
#            